### DISCUSSION TITLE
How to move stream and consumer replica?

### DISCUSSION THREAD

**Discussion by wade19870531** [nats-io/nats-server#3479] [General]
Sometimes servers NATS server are running on need to maintain or abnormal.
Is there any api for move replicas of stream/consumer to specific server or move away from a server?
I found this merge [https://github.com/nats-io/nats-server/pull/3217](url)
$JS.API.SERVER.STREAM.MOVE seem can move away from a server.
But I don't know how to execute? In the following, no response. Any document?
`nats request '$JS.API.STREAM.PEER.REMOVE' '{"peer":"nats-server-6-0"}'`

env:
nats 2.9.0
natscli: 0.0.34

**Comment by derekcollison** [comment]
If it is just one server and you have R>1 you can ask to have that peer removed from the system and the system will re-assign to other peers in the cluster. Meaning if you have a 3 node cluster should add a new peer first with a unique server name before removing the old peer.

`nats server raft peer-remove -h`

If you want to move an asset to another cluster, you can update the stream to different tags that will place it in the new cluster.

`nats stream update TEST_STREAM --tag cloud:gcp --tag geo:europe`

Example on how to move a stream in NGS from AWS in US to GCP in Europe.
